analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -92,45 +92,44 @@ private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
             {
                 checks.ExitReached(new(null, node.State));
             }
-            else if (node.Block.ContainsThrow())
-            {
-                yield return new(cfg.ExitBlock, node.State, null);
-            }
             else
             {
                 foreach (var successor in node.Block.Successors.Where(x => IsReachable(node, x)))
                 {
-                    if (ProcessBranch(node, successor) is { } newNode)
+                    foreach (var successorNode in ProcessBranch(node, successor).WhereNotNull())
                     {
-                        yield return newNode;
+                        yield return successorNode;
                     }
                 }
             }
         }
 
-        private ExplodedNode ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
+        private IEnumerable<ExplodedNode> ProcessBranch(ExplodedNode node, ControlFlowBranch branch)
         {
             if (branch.Destination is not null)
             {
-                return branch.FinallyRegions.Any() // When exiting finally region(s), redirect to 1st finally instead of the normal destination
+                yield return branch.FinallyRegions.Any() // When exiting finally region(s), redirect to 1st finally instead of the normal destination
                     ? FromFinally(new FinallyPoint(node.FinallyPoint, branch))
                     : CreateNode(branch.Destination, node.FinallyPoint);
             }
+            else if (node.Block.ContainsThrow())
+            {
+                foreach (var successor in ExceptionSuccessors(node, ExceptionCandidate(node.Block.BranchValue)))
```

Comment:
As discussed offline
```suggestion
                foreach (var successor in ExceptionSuccessors(node, ThrowExceptionType(node.Block.BranchValue)))
```